libs/blocks/global-navigation/global-navigation.js

```diff
@@ -465,7 +465,7 @@ class Gnav {
     const profileData = await fetch(`https://${env.adobeIO}/profile`, { headers });
 
     if (profileData.status !== 200) {
-      return;
+      throw new Error(`${profileData.statusText} url: ${profileData.url}`);
```

Comment:
If you think that we should log the error, I'd rather suggest to add a lana log than stopping the execution.

Reply:
You're right, the error would be caught - however I'd still suggest to stay consistent with the [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/fetch)

> A fetch() promise only rejects when the request fails, for example, because of a badly-formed request URL or a network error. A fetch() promise does not reject if the server responds with HTTP status codes that indicate errors (404, 504, etc.). Instead, a then() handler must check the [Response.ok](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) and/or [Response.status](https://developer.mozilla.org/en-US/docs/Web/API/Response/status) properties.

If a fetch fails because of a 404, we shouldn't throw an error in the method unless there is a legit error.
Also I'd assume that this could pollute the error logs even further, for something that sporadically happens.

My suggestion, and this is totally up to you / the tattva team:
```
    if (profileData.status !== 200) {
      lanaLog({ message: 'decorateProfile has failed to fetch profile data, e, tags: 'errorType=warn,module=gnav' });
      return
```

Reply:
IMO making the function fail, to trigger the method that calls `this.decorateProfile` to catch the error and hence log the lana log, is very error prone in case this behavior is ever refactored.

If you want to log something, you just log it - instead of throwing an error.

Reply:
Yes agree, throwing an error and making method fail will not be a good practice.
Added log near to the failure is the better way to handle this.